src/meta/src/barrier/mod.rs

```diff
@@ -564,6 +565,12 @@ where
         let mut barrier_timer: Option<HistogramTimer> = None;
         let (barrier_complete_tx, mut barrier_complete_rx) = tokio::sync::mpsc::unbounded_channel();
         let mut checkpoint_control = CheckpointControl::new(self.metrics.clone());
+        let (local_notification_tx, mut local_notification_rx) =
+            tokio::sync::mpsc::unbounded_channel();
+        self.env
+            .notification_manager()
+            .insert_local_sender(local_notification_tx)
+            .await;
```

Comment:
may off-topic: why don't we use `tokio::sync::broadcast` here? 🤔

Reply:
Maybe we were unaware of this crate at that time? cc @zwang28

Reply:
Anyway, we can decide which to use in another PR 🤣